### PR TITLE
Fix/redirect on delete

### PR DESCRIPTION
**Pull request by ivanzusko**
This **PR** does:
1. partially fixes this _issue_: https://github.com/hellofresh/janus-dashboard/issues/183.
2. fixes discarding search field after deleting the element.

### CODE REVIEW

File: src/__tests__/store/reducers/apiResponse.reducer.test.js [hunk 1]
```diff
@@ -75,12 +76,14 @@ describe('apiResponseReducer', () => {
         const apiName = getRandomString();
         const message = getRandomString();
         const onConfirm = getRandomString();
+        const shouldRedirect = getRandomString();
```
Comment (Vxplus) [inline src/__tests__/store/reducers/apiResponse.reducer.test.js]: why is it initialized as a string?, the name kind a implies a boolean val

File: src/__tests__/store/reducers/apiResponse.reducer.test.js [hunk 2]
```diff
@@ -98,6 +101,7 @@ describe('apiResponseReducer', () => {
                 expect(result.confirmationModal.api).toEqual(api);
                 expect(result.confirmationModal.apiName).toEqual(apiName);
                 expect(result.confirmationModal.message).toEqual(message);
+                expect(result.confirmationModal.shouldRedirect).toEqual(shouldRedirect);
```
Comment (Vxplus) [inline src/__tests__/store/reducers/apiResponse.reducer.test.js]: also I think if you need to check every value of the confirmation modal maybe use `toBe`, in the other case maybe just check the whole object.

File: src/modules/pages/EditPage/ApiItem.js [hunk 3]
```diff
@@ -41,8 +41,10 @@ class ApiItem extends PureComponent {
         getUpdatedEndpoint(values),
     )(this.props.selectedPlugins);
 
-    handleDelete = (apiName) => {
-        this.props.deleteEndpoint(apiName);
+    handleDelete = () => {
+        const needRedirect = true;
```
Comment (Vxplus) [inline src/modules/pages/EditPage/ApiItem.js]: why not just deleteEndpoint(this.props.api, true);?

File: src/store/actions/api.actions.js [hunk 4]
```diff
@@ -313,8 +313,8 @@ export const saveEndpoint = api => dispatch =>
 export const updateEndpoint = api => dispatch =>
     dispatch(openConfirmationModal('update', api, api.name));
 
-export const deleteEndpoint = apiName => dispatch =>
-    dispatch(openConfirmationModal('delete', {}, apiName));
+export const deleteEndpoint = (api, shouldRedirect/*: Boolean*/) => dispatch =>
```
Comment (Vxplus) [inline src/store/actions/api.actions.js]: why not `isRedirect`? also why not let the redirection be handle as a separate action?

File: src/__tests__/store/reducers/apiResponse.reducer.test.js (hunk 1 above)
Comment (ivanzusko) [reply src/__tests__/store/reducers/apiResponse.reducer.test.js]: in this particular case, I'm checking the existence and assigning of the property, so basically it doesn't matter. But agree, it's definitely better/consistent to work with proper type.

File: src/modules/pages/EditPage/ApiItem.js (hunk 3 above)
Comment (ivanzusko) [reply src/modules/pages/EditPage/ApiItem.js]: just for more explicit showing what exactly this `true` is doing.

File: src/store/actions/api.actions.js (hunk 4 above)
Comment (rafaeljesus) [reply src/store/actions/api.actions.js]: +1 for `isRedirect`

Comment (ivanzusko) [reply src/store/actions/api.actions.js]: 👨‍⚖️  : "The decision accepted!" 🔨

**Pull request merged**